doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,306 @@
+    },
+    {
+      "@type": "Property",
+      "name": "Bar",
+      "writable": true,
+      "schema": "string"
+    }
+  ]
+}
+```
+
+This model generates three Commands:
+
+| Command name | MQTT topic | Example request | Example response |
+| --- | --- | --- | --- |
+| ReadFoo | "sample/property/Foo/read" | | { "Foo": 33 } |
+| ReadBar | "sample/property/Bar/read" | | { "Bar": "hello" } |
+| WriteBar | "sample/property/Bar/write" | { "Bar": "bye" } | |
+
+The two read Commands have no request payloads.
+Their response payloads are objects, each with a single required field.
+The reverse holds for the write Command, which is defined only for the writable property.
+
+#### Basic sample model with aggregated Property topics
+
+The next sample model is identical to the above except for the property topic value, which does not contain a "{propertyName}" token:
+
+```json
+  "propertyTopic": "sample/property/{action}"
+```
+
+This model generates two Commands:
+
+| Command name | MQTT topic | Example request | Example response |
+| --- | --- | --- | --- |
+| Read | "sample/property/read" | | { "Foo": 33, "Bar": "hello" } |
+| Write | "sample/property/write" | { "Bar": "bye" } | |
+
+In general, a model that does not contain a "{propertyName}" token generates at most two Commands, one for reading and one for writing.
+If no Property in the Interface is writable, only a read Command is generated.
+
+The read Command has no request payload, and the write Command has no response payload.
+Fields in the read response are not optional and not nullable; all values are read on each read Command.
+Fields in the write request are optional; only fields that are present and have non-null values will be written.
+
+### New adjunct types: PropertyResult, PropertyValue, ReadError, and WriteError
+
+The second additive change to the Mqtt extension is support for modeling the error conditions that can result from a Property read or write.
+Mechanisms for modeling errors in Command responses are defined in [ADR 19](./0019-codegen-user-errs.md).
+The present ADR adds closely analogous mechanims for modeling errors in Property read/write responses.
+Specifically, the following four adjunct types are added.
+
+| New Adjunct Type | Material Cotype | Analogous ADR19 Type |
+| --- | --- | --- |
+| PropertyResult | Object | Result |
+| PropertyValue | Field | NormalResult |
+| ReadError | Field | ErrorResult |
+| WriteError | Field | ErrorResult |
+
+These adjunct types behave nearly identically, *mutatis mutandis*, to the analogous types defined for Command in ADR19.
+Specifically:
+
+* PropertyResult defines the type sent on the wire in a read response.
+* PropertyResult also defines the type sent on the wire in a write response, except there is no payload when there is no error.
+* PropertyValue defines the type returned by the Read API when there is no error.
+* PropertyValue also defines the type accepted by the Write API for writable Properties.
```

Comment:
isn't this slightly different since there's an extra layer of optionality on write requests vs read responses?

Reply:
Added asterisks and subsequent note.